src/actix/api/snapshot_api.rs

```diff
@@ -362,8 +362,8 @@ async fn upload_shard_snapshot(
 
         Result::<_, helpers::HttpError>::Ok(())
     })
-    .map_ok(|_| true)
-    .map_err(Into::into);
+    .map_err(Into::into)
+    .map(|res| res.and_then(|res| res));
```

Comment:
Too bad [`flatten`](https://doc.rust-lang.org/std/result/enum.Result.html#method.flatten) isn't stabilized yet :(

Reply:
We will, probably, get annoying `clippy` warning once it does. :D